packages/sdk-react/src/core/queries/useCounterpart.ts

```diff
@@ -502,48 +339,31 @@ export const useCounterpartContactById = (
   counterpartId: string,
   contactId?: string
 ) => {
-  const { monite } = useMoniteContext();
-  const { findById } = useCounterpartContactListCache(counterpartId);
-
-  return useQuery<CounterpartContactResponse | undefined, Error>({
-    queryKey: counterpartQueryKeys.contactDetail(counterpartId, contactId!),
-
-    queryFn: () => {
-      if (!contactId) return undefined;
-
-      const existedContact = findById(contactId);
-
-      if (existedContact) return existedContact;
-
-      return monite.api.counterparts.getContactById(counterpartId, contactId);
+  const { api } = useMoniteContext();
+
+  return api.counterparts.getCounterpartsIdContactsId.useQuery(
+    {
+      path: {
+        counterpart_id: counterpartId,
+        contact_id: contactId ?? '',
+      },
     },
-
-    enabled: Boolean(contactId) && Boolean(counterpartId),
-  });
+    {
+      enabled: !!counterpartId && !!contactId,
+    }
+  );
 };
 
-export const useUpdateCounterpartContact = (counterpartId: string) => {
+export const useUpdateCounterpartContact = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { invalidate } = useCounterpartListCache();
-  const { update } = useCounterpartContactListCache(counterpartId);
-
-  return useMutation<
-    CounterpartContactResponse,
-    Error,
-    CounterpartContactUpdate
-  >({
-    mutationFn: ({ contactId, payload }) =>
-      monite.api.counterparts.updateContact(counterpartId, contactId, payload),
-
-    onSuccess: (contact) => {
-      update(contact);
-      invalidate();
+  const { api } = useMoniteContext();
 
+  return api.counterparts.patchCounterpartsIdContactsId.useMutation(undefined, {
+    onSuccess: (counterpart) => {
```

Comment:
the endpoint responses with the `CounterpartContactResponse`, so, `contact` would be a better naming.